src/otx/v2/adapters/torch/mmengine/mmdet/modules/models/detectors/custom_atss_detector.py

```diff
@@ -0,0 +1,78 @@
+"""OTX ATSS Class for mmdetection detectors."""
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+from __future__ import annotations
+
+import functools
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from torch import nn
+
+from mmdet.models.detectors.atss import ATSS
+from mmdet.registry import MODELS
+
+from otx.v2.adapters.torch.modules.utils.task_adapt import map_class_names
+from otx.v2.api.utils.logger import get_logger
+
+logger = get_logger()
+
+
+@MODELS.register_module()
+class CustomATSS(ATSS):
+    """SAM optimizer & L2SP regularizer enabled custom ATSS."""
+
+    def __init__(self, *args, task_adapt: dict | None = None, **kwargs) -> None:
+        """Initialize CustomATSS detector."""
+        super().__init__(*args, **kwargs)
+
+        # Hook for class-sensitive weight loading
+        if task_adapt:
```

Comment:
What does `task_adapt` mean? I think we might need a different naming here.

Reply:
It means we will update model classes from classes of dataset and classes of model's checkpoint. 
I agree task_adapt is not straightforward, but it is used for a long time across tasks. 
If we want to change this name, then we should change for all at once.